R/module_nested_tabs.R

```diff
@@ -133,14 +133,14 @@ srv_nested_tabs <- function(id, datasets, modules, reporter = teal.reporter::Rep
 #' @rdname srv_nested_tabs
 #' @export
 #' @keywords internal
-srv_nested_tabs.default <- function(id, datasets, modules, reporter) {
+srv_nested_tabs.default <- function(id, datasets, modules, reporter = teal.reporter::Reporter$new()) {
```

Comment:
I don't think we need to define the default value during the method dispatch because it's being specified during the creation of the generic function here:
https://github.com/insightsengineering/teal/blob/e8a67d1f623c16b101b70ddd39caae2426d6d2e3/R/module_nested_tabs.R#L127-L130

However, I did look at `ui_nested_tabs` where it defines the default value during dispatch for `depth`.
For consistency, I'm fine with adding the default value during method dispatch.

Reply:
It's actually needed here otherwise we get an error. I don't think this is because S3 though. I think it's with how `srv_nested_tabs.teal_modules` works.  It uses the following code ([here](https://github.com/insightsengineering/teal/blob/e8a67d1f623c16b101b70ddd39caae2426d6d2e3/R/module_nested_tabs.R#L153-L155))

```r
    modules_reactive <- sapply(names(modules$children), USE.NAMES = TRUE, function(id) {
      srv_nested_tabs(id = id, datasets = datasets, modules = modules$children[[id]], reporter = reporter)
    })
```

In the call to `srv_nested_tabs`, `id` is a string. There is no method for this so it reverts back to the generic. Then it hits `inherits(reporter, "Reporter")` and it fails. There's a recursiveness here generic -> method -> generic that I think is throwing things off.

Reply:
#822